test/test_autograd.py

```diff
@@ -8460,6 +8460,93 @@ def _test_op(fn, inp, args):
         _test_op(torch.view_as_complex, torch.rand(2, 2), ())
         _test_op(torch.view_as_real, torch.rand(2, 2, dtype=torch.cfloat), ())
 
+    def test_forward_has_default_parameters(self):
```

Comment:
Maybe rename this test to something specific?
`test_setup_context_when_forward_has_default_args`
And since we changed it to no longer affect the non-setup-context case, maybe we don't need the rest of the tests?

Reply:
And isn't the most straightforward test here is just to literally inspect the inputs of setup_context and assert that it is what you expect. Everything else high-level like the gradients should just follow from that.

Reply:
Done